packages/cloud_firestore/cloud_firestore/ios/Classes/FLTFirebaseFirestoreReader.m

```diff
@@ -95,14 +95,20 @@ - (FIRFirestoreSettings *)FIRFirestoreSettings {
 
   if (![values[@"persistenceEnabled"] isEqual:[NSNull null]]) {
     bool persistEnabled = [((NSNumber *)values[@"persistenceEnabled"]) boolValue];
-    int size = [((NSNumber *)values[@"cacheSizeBytes"]) intValue];
+
+    // This is the maximum amount of cache allowed. We use the same number on android.
+    // This now causes an exception: kFIRFirestoreCacheSizeUnlimited
+    NSNumber *size = @104857600;
+
+    if (![values[@"cacheSizeBytes"] isEqual:[NSNull null]]) {
+      NSNumber *cacheSizeBytes = ((NSNumber *)values[@"cacheSizeBytes"]);
+      if ([cacheSizeBytes intValue] != -1) {
+        size = cacheSizeBytes;
+      }
+    }
 
     if (persistEnabled) {
-      NSNumber *unlimitedPersistence =
-          [NSNumber numberWithLongLong:kFIRFirestoreCacheSizeUnlimited];
```

Comment:
Was this constant created by us or was it exposed by the Firebase SDK? Should we ask the Firebase iOS SDK explanations about it if it's exposed by the native sdk?

Reply:
Yeah, it's one of the SDKs constants. Worth asking in the next meeting 👍